Fix NAV_STATE enum values

fiam opened the pull request:
Since POSHOLD_2D entries were removed, we need to add 2 dummy
entries to keep the old values as they were, since they are logged
to blackbox.

fiam committed:
Fix NAV_STATE enum values

Since POSHOLD_2D entries were removed, we need to add 2 dummy
entries to keep the old values as they were, since they are logged
to blackbox.

digitalentity reviewed (commented):
Each navigation fsm state consumes flash since it's an array entry. However, we might reuse these entries for i.e. @giacomo892's CRUISE mode ;)

fiam committed:
Add a new field to each navigation state with its public ID

This ID will be permanent from now on, so we cahn freely change
navigationFSMState_t without having to adjust external logs
dealing with blackbox logs.

digitalentity commented on src/main/navigation/navigation_private.h:
```diff
@@ -176,41 +176,82 @@ typedef enum {
 } navigationFSMEvent_t;
 
+// This enum is used to keep values in blackbox logs stable, so we can
+// freely change navigationFSMState_t.
```
I don't like the name `PUBLIC_STATE`, same naming as in box modes seems more reasonable - `PERSISTENT_ID`

fiam replied:
Roger that

digitalentity commented on src/main/navigation/navigation_private.h:
```diff
@@ -243,6 +284,7 @@ typedef enum {
 } navigationFSMStateFlags_t;
 
 typedef struct {
+    navigationPublicState_t             publicState;
```
`persistentId` ?

fiam committed:
Rename from "public state" to "persistent ID"

- NAV_PUBLIC_STATE_* -> NAV_PERSISTENT_ID_*
- .publicState -> .persistentId
- .navPublicState -> .navPersistentId
